salt/utils/event.py

```diff
@@ -529,8 +529,8 @@ def _get_event(self, wait, tag, match_func=None, no_block=False):
                 ret = {'data': data, 'tag': mtag}
             except KeyboardInterrupt:
                 return {'tag': 'salt/event/exit', 'data': {}}
-            except (tornado.iostream.StreamClosedError, RuntimeError):
-                return None
+            except (tornado.iostream.StreamClosedError, RuntimeError) as exc:
+                raise exc
```

Comment:
Thinking about this further, I would also not throw an exception is `self._closing` is True.